accepted/00000-hub-reporting.md

```diff
@@ -0,0 +1,177 @@
+aggregates them. Other databases with the same schema are also present in all the server, to collect 
+information for that system.
+All the databases (in the Uyuni Hub and in all Uyuni Server) needs to be made available on the network to either connect in a secure way (using the SSL
+certificates provided by Uyuni Server) with the reporting tool or (for Uyuni Server) with the Hub to gather the data. 
+If possible the connection to the main Uyuni DB from the outside should be forbidden.
+In summary:
+- DB in Uyuni Hub stores, collects and eventually aggregates data coming from all the DB in Uyuni Server,
+- DB in Uyuni Server stores it own data,
+- The reporting tool can be connected either to the Hub or to any Uyuni Server,
+- all connections are secure.
+
+### The Database Schema
+
+The schema should export the most important tables from the main Uyuni Database as a slightly de-normalized
+variant containing only data which are relevant for a report.
+
+A ready-to-use report can be provided with views doing joins over multiple tables.
+No foreign keys should be used to make data update and refreshing easier and independent from the order
+of the tables.
+
+Every table gets an extra column for the Uyuni server id of server which provide the data (`mgm_id`).
+On a single Uyuni Server this column has the standard value `1` which represent "localhost".
+On the Hub it will be replaced with the real server id the managed server has in the hub database.
+
+Every row should get a report timestamp column (`exported`), which is set to the time when the data
+were exported from the main Uyuni Server database.
+
+Indexes should be set on the typical columns which might be searched for.
+
+Data which belong to the traditional stack only (like osad status) should not be made available in the
+reporting database as the traditional stack will be deprecated soon.
+
+A detailed Database schema is not part of this RFC.
+
+# Installation
+The reporting tools will be installed by default; all the upgrade to a newer version, will install the 
+reporting tools as well. Several packages will be provided in order to provide the following tools and
+functionalities:
+- setup the database and initialize the schema,
+- perform database schema migration,
+- setup database users,
+- setup tasktomatic jobs.
+
+All the other information required (e.g. Uyuni Server info) will be retrieved by already existing configuration.
+
+## Reporting Database
+A reporting DB is setup in all Uyuni Servers and Uyuni Hub nodes to store reporting data.
+The default setup will create a Read/Write user, to manage the database and write the data from it into the Reporting DB.
+This user and the DB connection parameters are written into `/etc/rhn/rhn.conf` similar to the default DB options.
+
+## Initial Setup
+Uyuni Hub should create its own read-only user on the Reporting Databases of the single Servers.
+As the Servers are managed with salt, we will write a state to create an account.
```

Comment:
What pros and cons do you see compared to a solution in which the read-only user is automatically created on each Server, and then those credentials are collected via Salt by the Hub?

Reply:
If we create the password on the hub, we send it only via secured pillar data to the server and it does not need to be stored in plain text on the single server. Only the R/W credentials used on the server itself are stored in the config.
Also credentials the admin creates for his reporting tool do not need to be stored in any kind of config on the server.

I will add this to the text.